fabric-item-api-v1/src/main/java/net/fabricmc/fabric/impl/item/ShearsHelper.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.item;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.item.ShearsItem;
+import net.minecraft.predicate.item.ItemPredicate;
+import net.minecraft.registry.Registries;
+import net.minecraft.registry.entry.RegistryEntry;
+import net.minecraft.registry.tag.TagKey;
+
+import net.fabricmc.api.ModInitializer;
+import net.fabricmc.fabric.api.event.lifecycle.v1.CommonLifecycleEvents;
+import net.fabricmc.fabric.impl.tag.convention.TagRegistration;
+import net.fabricmc.fabric.mixin.item.shears.accessors.DirectRegistryEntryListAccessor;
+
+public final class ShearsHelper implements ModInitializer {
+	public static final TagKey<Item> FABRIC_SHEARS = TagRegistration.ITEM_TAG_REGISTRATION.registerFabric("shears");
+	public static final List<ItemPredicate> MATCH_TOOL_PREDICATES = new ArrayList<>();
+	private static final Set<RegistryEntry<Item>> SHEARS_ITEMS = new HashSet<>();
+
+	public static boolean isShears(ItemStack stack, Item item) {
```

Comment:
Whats the need for this, the ItemStack contains the item.

Reply:
it also checks the `item` argument is `Items.SHEARS` just in case because there are more than just the shears checks using `ItemStack#isOf` in the mixins (it shouldnt even matter since the shears checks are always first, but who knows)